src/tmt_web/api.py

```diff
@@ -122,49 +184,120 @@ def find_test(
         response = response_by_output.get(out_format, PlainTextResponse)
         return response(service.main(**service_args))
 
+    logger.debug("Processing request asynchronously with Celery")
     r = service.main.delay(**service_args)
 
-    # Special handling of response if the format is html
-    # TODO: Shouldn't be the "yaml" format also covered with a `PlainTextResponse`?
+    # Handle response based on format
     if out_format == "html":
+        logger.debug("Generating HTML status callback")
         status_callback_url = f"{settings.API_HOSTNAME}/status/html?task-id={r.task_id}"
-        return HTMLResponse(content=html_generator.generate_status_callback(r, status_callback_url))
-
+        return HTMLResponse(
+            content=html_generator.generate_status_callback(r, status_callback_url, logger),
+        )
+    if out_format == "yaml":
+        task_out = _to_task_out(r)
+        return PlainTextResponse(content=task_out.model_dump_json())
```

Comment:
`format=yaml` actually returns json:
```
❯ http "http://localhost:8000/?plan-url=https://gitlab.com/testing-farm/tests&plan-name=/testing-farm/sanity&format=yaml"
HTTP/1.1 200 OK
content-length: 176
content-type: text/plain; charset=utf-8
date: Tue, 10 Dec 2024 12:50:58 GMT
server: uvicorn

{
    "id": "50c52b5e-16e7-49a7-abac-8671e9ed8f42",
    "result": null,
    "status": "PENDING",
    "status_callback_url": "http://localhost:8000/status?task-id=50c52b5e-16e7-49a7-abac-8671e9ed8f42"
}
```

Reply:
I've briefly looked at this, definitely a bug, but do we want yaml also for the pending state?  Also, once ready, only the yaml should be returned, right?

Reply:
How about now? :)

Reply:
completely reworked where and how we're doing the formatting. Could you please give it a go @janhavlin ? :)